packages/react-scripts/per-lang-loader/index.js

```diff
@@ -0,0 +1,149 @@
+'use strict';
+// This file is mostly copied from https://github.com/alienfast/i18next-loader/blob/develop/index.js
+// and adjusted to our needs for per-lang Bundling
+
+const path = require('path');
+const fs = require('fs');
+const globAll = require('glob-all');
+const loaderUtils = require('loader-utils');
+
+function enumerateLangs(dir) {
+  return fs.readdirSync(dir).filter(function(file) {
+    return fs.statSync(path.join(dir, file)).isDirectory();
+  });
+}
+
+//https://github.com/jpillora/node-glob-all#usage
+function findAll(globs, cwd) {
+  const globArray = Array.isArray(globs) ? globs : [globs];
+  return globAll.sync(globArray, { cwd, realpath: true });
+}
+
+const i18nextImport = `
+import i18n from "i18next";
+
+const handleLoadLocale = (locale, namespace, localeStrings) => {
+  i18n.addResources(locale, namespace, localeStrings)
+}
+const handleError = (locale, namespace) => {
+  console.error(\`failed to load translation - \${locale}" + "\${namespace}"\`)
+}
+const importLocale = (locale, namespace) => {
+  import(\`./\${locale}/\${namespace}\`)
+    .then(({ default: localeStrings }) => handleLoadLocale(locale, namespace, localeStrings))
+    .catch(() => handleError(locale, namespace))
+}
+
+`;
+
+// the mainExport is just for backwards compatability.
+// Files call import translations from './locales'. Without
+// a default export, that breaks.
+const mainExport = 'export default {}';
+
+const createDynamicImport = ns =>
+  `
+if(i18n.language){
+  importLocale(i18n.language, '${ns}');
+} else {
+  i18n.on('languageChanged', locale => {
+    importLocale(locale, '${ns}')
+  });
+}
+
+// Always load english translations for backup.
+importLocale('en', '${ns}');
+
+`;
+
+module.exports = function(source) {
+  this.cacheable && this.cacheable();
+
+  // if json file, just load it
+  if (this.resource.endsWith('.json')) {
+    return source;
+  }
```

Comment:
So, it seems like we don't want to run this with any json files. Help me understand, are we wanting/expecting this to only be run 1 time for the `{appPath}/locales/index.js` file? 

Are there potentially other .js files in the locales directory? If so, will this script be run multiple times then? Should we "beef up" the webpack [rule's `test` here](https://github.com/fs-webdev/create-react-app/pull/288/files/e9676f6e17c9ab4f993c56851e31962348f0bbbf#diff-8e25c4f6f592c1fcfc38f0d43d62cbd68399f44f494c1b60f0cf9ccd7344d697R413) so it only gets run 1 time total for the app?

Reply:
Yeah, we are including the json files because of the test rule we have. I'll change it.